diff --git a/core/lib/oauth/client.js b/core/lib/oauth/client.js
index 77161bd8ce972a50611bf3744d59fadf33d02955..2607526a62b73753b5a310f79ac82c3ceae32a43 100644
--- a/core/lib/oauth/client.js
+++ b/core/lib/oauth/client.js
@@ -6,10 +6,16 @@ Object.defineProperty(exports, "__esModule", {
 exports.openidClient = openidClient;
 
 var _openidClient = require("openid-client");
-
+var HttpsProxyAgent = require("https-proxy-agent");
 async function openidClient(options) {
   const provider = options.provider;
-  if (provider.httpOptions) _openidClient.custom.setHttpOptionsDefaults(provider.httpOptions);
+  let httpOptions = {};
+  if (provider.httpOptions) httpOptions = { ...provider.httpOptions };
+  if (process.env.PROXY) {
+    let agent = new HttpsProxyAgent(process.env.PROXY);
+    httpOptions.agent = agent;
+  }
+  _openidClient.custom.setHttpOptionsDefaults(httpOptions);
   let issuer;
 
   if (provider.wellKnown) {