diff --git a/typings/react-flip-move.d.ts b/typings/react-flip-move.d.ts
index 6e8a1b8e6dd0cd2f612b5a0ee72b93682060ca2b..561158a7c650fd76838e9a91b5215cf3a3a44bc7 100644
--- a/typings/react-flip-move.d.ts
+++ b/typings/react-flip-move.d.ts
@@ -26,6 +26,7 @@ declare namespace FlipMove {
     type AnimationProp = AnimationPreset | Animation | boolean;
 
     interface FlipMoveProps {
+        children?: React.ReactNode;
         className?: string;
 
         /**